types/src/transaction/script.rs

```diff
@@ -50,3 +51,80 @@ impl fmt::Debug for Script {
             .finish()
     }
 }
+
+/// How to call a particular Move script (aka. an "ABI").
+#[derive(Clone, Debug, Hash, Eq, PartialEq, Serialize, Deserialize)]
+pub struct ScriptABI {
+    /// The public name of the script.
+    name: String,
+    /// Some text comment.
+    doc: String,
+    /// The `code` value to set in the `Script` object.
+    code: Vec<u8>,
+    /// The names of the type arguments.
+    ty_args: Vec<String>,
+    /// The description of regular arguments.
+    args: Vec<ArgumentABI>,
+}
+
+/// The description of a script argument.
+#[derive(Clone, Debug, Hash, Eq, PartialEq, Serialize, Deserialize)]
+pub struct ArgumentABI {
+    /// The name of the argument.
+    name: String,
```

Comment:
Same as above, I'm not sure if the name will always be available.

Reply:
It is available now and we need it on the client side.

Reply:
It's true that we can't scrape these names from bytecode (if we go that route in the future), but seems nice to provide them if we're starting from Rust instead.

Reply:
That's interesting. I'm not that familiar with the background but I assume this ABI is generated from the Move source files rather than the binary file format?

Reply:
Sam convinced me today that these ABIs will likely (1) not be stored onchain (2) be managed by the author of the source code (and made available in the same adhoc way).

Reply:
@vgao1996: it's hard coded now in the description of the script as in files above